[2.19] Clarify `library` usage for no-name declarations update (#4390)
dart-lang/site-www: src/_guides/libraries/create-library-packages.md

For an example of generated docs, see the
[shelf documentation.]({{site.pub-api}}/shelf/latest)

To include any *library-level* documentation in the generated docs,

> you must specify the `library` directive and attach the comment directly above it.

Reviewer: Can we simplify this?

```suggestion
add a `library` directive and attach the comment directly above it.
```

{{site.alert.note}}
  To include any library-level documentation in the generated docs,
  you must specify the `library` directive and attach the comment directly above it.

>   See [Effective Dart: Documentation](/guides/language/effective-dart/documentation#consider-writing-a-library-level-doc-comment)

Reviewer: I think now that this isn't talking about an exceptional situation, perhaps this should no longer be a note, but rather just part of the text. What do you think?

Also for the Effective Dart link, you should tell them why to visit the link or what they will find there.

Reviewer: What might be useful as a note though is a version-note saying unnamed libraries require a [language version](https://dart.dev/guides/language/evolution#language-versioning) of at least 2.19.

Author: > What might be worth considering as a note though is a version-note saying unnamed libraries require a [language version](https://dart.dev/guides/language/evolution#language-versioning) of at least 2.19.

I think the pointer to Effective Dart is good enough here, because this text doesn't mention unnamed libraries so bringing it up is a little off topic.
